adr/0003-jsonpath-support.md

```diff
@@ -0,0 +1,43 @@
+# 3. JSONPath support
+
+Date: 15-03-2024
+
+## Status
+
+Draft
+
+## Context
+
+Support for JsonPATH was requested by some potential library clients.  
+However, JsonPATH RFC specifies way too many features which are tedious to implement without having runtime dependencies.  
+Therefore, we have to decide which features are necessary and which are not. 
```

Comment:
The reason we don't support all JsonPath features is actually not because it is tedious to implement, but there are three other main reasons:

- First of all, JsonPath support was added to be able to map Java class fields to JsonPaths because if you would have the same field nested in a different field it is impossible to distinguish the two without JsonPath. The limited JsonPath we currently support is precisely what is required to fully support this use case (e.g. works both with collection fields as with nested objects).
- Second, with every feature we add to the library, we add maintenance complexity and the probably to introduce bugs. Since no-one requested any of the other JsonPath features, there is no point to add it risking us to introduce a bug besides the currently supported JsonPath features.
- Finally, with every feature we add to the library performance usually gets slightly worse which again, is not worth it considering no-one ever requested any other JsonPath feature besides the ones you implemented so far :-)

Reply:
Tedious to implement is basically not a reason, because if that would be the case we could just open an issue and ask the community to implement it or do it ourselves on a rainy Sunday :-)

Reply:
Also I would add that JSONPath was made first and foremost for selecting values, but some things simply don't make sense for masking (reasons for ambiguity checks for example)